src/Symfony/Component/Yaml/Parser.php

```diff
@@ -230,8 +230,12 @@ private function doParse($value, $flags)
                         $this->parseBlock($this->getRealCurrentLineNb() + 1, $this->getNextEmbedBlock(null, true), $flags)
                     );
                 } else {
-                    if (isset($values['leadspaces'])
-                        && self::preg_match('#^(?P<key>'.Inline::REGEX_QUOTED_STRING.'|[^ \'"\{\[].*?) *\:(\s+(?P<value>.+?))?\s*$#u', $this->trimTag($values['value']), $matches)
+                    if (
+                        isset($values['leadspaces'])
+                        && (
+                            0 === strpos($values['value'], '!php/const')
```

Comment:
I wonder if we should consider other tags (like `!!float`) too.

Reply:
Test still pass if the statement here just checks for a starting '!' instead of  '!php/const'. That should allow other tags in addition to PHP constants as keys.